### PR TITLE
package - Add rename-package

### PR DESCRIPTION
**Pull request by sheepduke**
Adds some introduction to `RENAME-PACKAGE`.

### CODE REVIEW

File: packages.md [hunk 1]
```diff
@@ -35,6 +35,26 @@ Or you can do it with [`LOOP`][loop].
   collect s)
 ~~~
 
+# Give package local nickname
```
Comment (vindarel) [inline packages.md]: give *a* package *a* local nickname

File: packages.md [hunk 2]
```diff
@@ -35,6 +35,26 @@ Or you can do it with [`LOOP`][loop].
...
+
+Sometimes it is handy to give imported a local name for saving some typing.
```
Comment (vindarel) [inline packages.md]: IMO:

sometimes it is handy to give a local name to an imported package to save some typing.

File: packages.md [hunk 3]
```diff
@@ -35,6 +35,26 @@ Or you can do it with [`LOOP`][loop].
...
+This can be achieved by using [`RENAME-PACKAGE`][rename-package]. For example:
+
+~~~lisp
+(asdf:load-system :cl-ppcre)
+
+(defpackage :mypackage
+  (:use :cl))
+(in-package :mypackage)
+
+(rename-package :cl-ppcre :re)
```
Comment (vindarel) [inline packages.md]: Also a note about `:nickname` in package definitions ?

Comment (vindarel) [reply packages.md]: maybe my remark is a better fit for systems.md. (do as you like)

File: packages.md (hunk 2 above)
Comment (sheepduke) [reply packages.md]: To be honest these contents was finished before a meeting so I was really in hurry.

I apologize for not double-checking before committing.

File: packages.md (hunk 3 above)
Comment (sheepduke) [reply packages.md]: I think might be better to clearify that "nickname" mentioned here is different with the one in `DEFPACKAGE`.

I will add it.

File: packages.md (hunk 1 above)
Comment (sheepduke) [reply packages.md]: Fixed. Please check.

File: packages.md (hunk 2 above)
Comment (sheepduke) [reply packages.md]: Fixed. Please check.

File: packages.md (hunk 3 above)
Comment (sheepduke) [reply packages.md]: Fixed. Please check.

File: packages.md [hunk 4]
```diff
@@ -54,7 +78,58 @@ This can be achieved by using [`RENAME-PACKAGE`][rename-package]. For example:
 (re:scan "a" "abc")
 ~~~
 
+Please note that the original name (`:CL-PPCRE` in the example above) will
+**not** be available any more after renaming.
+
+## Package locks
+
+A package can be declared to be locked so that it cannot be modified by the
+user. Attempts to change its symbol table or redefine functions which its
```
Comment (vindarel) [inline packages.md]: As a reader I ask myself: but why ? There's an answer in the SBCL doc:

> The package common-lisp and SBCL internal implementation packages are locked by default, including sb-ext. 

shall we (you :) ) add this explanation ?

**Comment by vindarel** [comment]
Nice addition ! Another suggestion inside.

File: packages.md [hunk 5]
```diff
@@ -35,6 +35,101 @@ Or you can do it with [`LOOP`][loop].
   collect s)
 ~~~
 
+# Package nickname
+
+## Nickname Provided by Packages
+
+When defining a package, it is normal to give it a nickname for better user
```
Comment (pnathan) [inline packages.md]: I would note that that overuse of package aliases can wind up extremely confusing for future maintainers.

**Comment by pnathan** [comment]
Just one comment, regarding maintainability concerns.

File: packages.md (hunk 5 above)
Comment (sheepduke) [reply packages.md]: As a man who has a bunch of nicknames, I would say "it depends".

If the nickname are meaningful and straightforward, I think it is OK. Also TBH I seldom see a package with more than one nickname.

But you are right. Giving too many nicknames to a package may lead to chaos especially when people use different nicknames simultaneously.

I will add some explanation of this. Thanks for your advice.

**Comment by pnathan** [comment]
LGTM

**Comment by sheepduke** [comment]
> Thanks, and pleasure to see a new contributor.

Welcome myself ;-)

**Pull request merged**